src/main/scala/com/github/unchama/seichiassist/subsystems/vote/subsystems/fairy/bukkit/actions/BukkitRecoveryMana.scala

```diff
@@ -18,223 +16,219 @@ import com.github.unchama.seichiassist.subsystems.vote.subsystems.fairy.domain.p
-                       Random.nextInt(reflectedMineStackedAmount / 50)
-                     else 0)
-                }
-
-                // minestackからりんごを消費する
-                playerdata
-                  .minestack
-                  .subtractStackedAmountOf(gachaRingoObject, appleConsumptionAmount)
-
-                // 消費したりんごの量を保存する
-                fairyAPI
-                  .increaseAppleAteByFairy(uuid, AppleAmount(appleConsumptionAmount.toInt))
-                  .toIO
-                  .unsafeRunSync()
-
-                // マナを回復する
-                ContextCoercion(
-                  manaApi.manaAmount(player).restoreAbsolute(ManaAmount(recoveryManaAmount))
-                ).toIO.unsafeRunSync()
-
-                if (finallyAppleConsumptionAmount > mineStackedGachaRingoAmount) {
-                  MessageEffect(s"$RESET$YELLOW${BOLD}MineStackにがちゃりんごがないようです。。。")
-                    .apply(player)
-                    .unsafeRunSync()
-                }
-
-                SequentialEffect(
-                  UnfocusedEffect {
-                    new FairySpeech[F, G]
-                      .speechRandomly(
-                        player,
-                        if (finallyAppleConsumptionAmount > mineStackedGachaRingoAmount)
-                          FairyManaRecoveryState.notConsumptionApple
-                        else FairyManaRecoveryState.consumptionApple
-                      )
-                      .toIO
-                      .unsafeRunSync()
-                  },
-                  MessageEffect(s"$RESET$YELLOW${BOLD}マナ妖精が${recoveryManaAmount}マナを回復してくれました"),
-                  if (appleConsumptionAmount != 0)
-                    MessageEffect(
-                      s"$RESET$YELLOW${BOLD}あっ！${appleConsumptionAmount}個のがちゃりんごが食べられてる！"
-                    )
-                  else MessageEffect(s"$RESET$YELLOW${BOLD}あなたは妖精にりんごを渡しませんでした。")
-                ).apply(player).unsafeRunSync()
-              }
-              eff.toIO.unsafeRunSync()
-            }
+              else MessageEffect(s"$RESET$YELLOW${BOLD}あなたは妖精にりんごを渡しませんでした。")
+            ).apply(player)
           }
-        } else Sync[F].unit
-      }
+      }.whenA(isUsing && !oldManaAmount.isFull)
+    } yield ()
+
+  /**
+   * MineStackに入っているがちゃりんごの数を考慮していない
+   * がちゃりんごの消費量を計算します。
+   */
+  private def computeAppleConsumptionAmount: F[Int] = for {
+    seichiAmountData <- ContextCoercion(breakCountAPI.seichiAmountDataRepository(player).read)
+    chainVoteNumber <- voteAPI.chainVoteDayNumber(uuid)
+    appleOpenState <- fairyAPI.appleOpenState(uuid)
+    oldManaAmount <- ContextCoercion {
+      manaApi.readManaAmount(player)
+    }
+  } yield {
+    val playerLevel = seichiAmountData.levelCorrespondingToExp
+
+    val isAppleOpenStateIsOpenOrOpenALittle =
+      appleOpenState == AppleOpenState.OpenALittle || appleOpenState == AppleOpenState.Open
+    val isManaExistsSeventyFivePercent = oldManaAmount.ratioToCap.exists(_ >= 0.75)
```

Comment:
```suggestion
    val isEnoughMana = oldManaAmount.ratioToCap.exists(_ >= 0.75)
```

ローカル変数なので実装詳細を説明するほどの具象的な名前をつける必要はありません